plugin/trino-redshift/src/test/java/io/trino/plugin/redshift/TestRedshiftUnload.java

```diff
@@ -0,0 +1,106 @@
+import static io.trino.plugin.redshift.RedshiftQueryRunner.AWS_SECRET_KEY;
+import static io.trino.plugin.redshift.RedshiftQueryRunner.IAM_ROLE;
+import static io.trino.plugin.redshift.RedshiftQueryRunner.S3_UNLOAD_ROOT;
+import static io.trino.tpch.TpchTable.NATION;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.TestInstance.Lifecycle.PER_CLASS;
+import static org.junit.jupiter.api.parallel.ExecutionMode.CONCURRENT;
+
+@TestInstance(PER_CLASS)
+@Execution(CONCURRENT)
+final class TestRedshiftUnload
+        extends AbstractTestQueries
+{
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        return RedshiftQueryRunner.builder()
+                .setConnectorProperties(
+                        Map.of(
+                                "redshift.unload-location", S3_UNLOAD_ROOT,
+                                "redshift.unload-iam-role", IAM_ROLE,
+                                "s3.region", AWS_REGION,
+                                "s3.endpoint", "https://s3.%s.amazonaws.com".formatted(AWS_REGION),
+                                "s3.aws-access-key", AWS_ACCESS_KEY,
+                                "s3.aws-secret-key", AWS_SECRET_KEY,
+                                "s3.path-style-access", "true",
+                                "join-pushdown.enabled", "true",
+                                "join-pushdown.strategy", "EAGER"))
+                .setInitialTables(List.of(NATION))
+                .build();
+    }
+
+    @Test
+    public void verifyUnloadEnabled()
+    {
+        assertQuery(
+                "SHOW SESSION LIKE 'redshift.unload_enabled'",
+                "VALUES ('redshift.unload_enabled', 'true', 'true', 'boolean', 'Use UNLOAD for reading query results')");
+    }
+
+    @Test
+    void testUnloadFlow()
+    {
+        assertQueryStats(
+                getSession(),
+                "SELECT nationkey, name FROM nation where regionkey = 0",
+                queryStats -> {
+                    Map<String, String> splitInfo =
+                            ((SplitOperatorInfo) queryStats.getOperatorSummaries()
+                                    .stream()
+                                    .filter(summary -> summary.getOperatorType().startsWith("TableScanOperator"))
+                                    .collect(onlyElement())
+                                    .getInfo())
+                                    .getSplitInfo();
+                    assertThat(splitInfo.get("path")).matches("%s/.*/0001_part_00.parquet".formatted(S3_UNLOAD_ROOT));
+                },
+                results -> assertThat(results.getRowCount()).isEqualTo(5));
+    }
+
+    @Test
+    void testUnloadFlowFallbackToJdbc()
+    {
+        // Fallback to JDBC as limit clause is not supported by UNLOAD
+        assertQueryStats(
+                getSession(),
+                "SELECT nationkey, name FROM nation where regionkey = 0 limit 2",
```

Comment:
https://docs.aws.amazon.com/redshift/latest/dg/r_UNLOAD.html contains inner LIMIT example. Is it possible to support LIMIT clause by adjusting the generated SELECT query? e.g. `SELECT * FROM (SELECT * FROM region LIMIT 1)`

Reply:
yes its possible to support limit in the next iteration but let's limit the scope of this PR.

Reply:
Please file an issue and leave TODO comment.

Reply:
sure created https://github.com/trinodb/trino/issues/24480 and added a code comment in RedshiftSplitManager

Reply:
I'm not sure that we should support LIMIT. Since we know the output is likely to be small it may be better to skip S3 round trip and read from JDBC directly. Maybe we could support it for large LIMIT N but we should evaluate that by benchmarking some queries first.